homeassistant/components/bluetooth/passive_update_processor.py

```diff
@@ -644,6 +645,7 @@ def __init__(
             self._attr_unique_id = f"{address}-{key}"
         if ATTR_NAME not in self._attr_device_info:
             self._attr_device_info[ATTR_NAME] = self.processor.coordinator.name
+        self._attr_device_info[ATTR_CONNECTIONS] = {(CONNECTION_BLUETOOTH, address)}
```

Comment:
I'm not sure we can do this as it will merge secondary devices together. See above for how identifiers is used.

Reply:
So I should do this as with the identifiers?

Reply:
I don't think it's permissible to add a trailer string to the MAC since we expect it's formatted in a specific manner which is why it's an identifier here (there are no formatting rules other than string)

I think you can safely add the Mac if the device_id is None which indicates it's the primary device.  That should still solve the linked issue without merging the devices unexpectedly.

Reply:
Thanks, learned something new :) I've added the guard.